zebra-chain/src/transaction.rs

```diff
@@ -540,4 +542,147 @@ impl Transaction {
         self.orchard_shielded_data()
             .map(|orchard_shielded_data| orchard_shielded_data.flags)
     }
+
+    // value pool
+
+    /// Get all the value pools for this transaction
+    pub fn value_balance(
+        &self,
+        utxos: &HashMap<transparent::OutPoint, transparent::OrderedUtxo>,
+    ) -> Result<ValueBalance<NegativeAllowed>, Box<dyn std::error::Error>> {
+        match self {
+            Transaction::V1 {
+                inputs, outputs, ..
+            } => {
```

Comment:
Can you implement this method using existing `Transaction` methods like `inputs()`?

If we access each field in each transaction version separately, the code is longer and harder to review. And we'll need tests for each different transaction version.